detox/android/detox/src/full/java/com/wix/detox/espresso/action/DetoxAccessibilityAction.java

```diff
@@ -0,0 +1,59 @@
+package com.wix.detox.espresso.action;
+
+import android.view.View;
+
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.uimanager.UIManagerHelper;
+import com.facebook.react.uimanager.UIManagerModule;
+import com.facebook.react.uimanager.events.Event;
+import com.facebook.react.uimanager.events.RCTEventEmitter;
+import com.wix.detox.espresso.DetoxMatcher;
+
+import org.hamcrest.Matcher;
+
+public class DetoxAccessibilityAction implements ViewAction {
+    String mActionName;
+
+    public DetoxAccessibilityAction(String actionName) {
+        mActionName = actionName;
+    }
+
+    @Override
+    public Matcher<View> getConstraints() {
+        return DetoxMatcher.matcherForNotNull();
+    }
+
+    @Override
+    public String getDescription() {
+        return "Dispatch an Accessibility Action";
+    }
+
+    @Override
+    public void perform(UiController uiController, View view) {
+        ReactContext context = (ReactContext) view.getContext();
+        final int reactTag = view.getId();
+        final WritableMap event = Arguments.createMap();
+        event.putString("actionName", mActionName);
+
+        UIManagerModule uiManager = (UIManagerModule) UIManagerHelper.getUIManager(context, reactTag);
+        uiManager.getEventDispatcher().dispatchEvent(new Event(reactTag) {
+            @Override
+            public String getEventName() {
+                return "topAccessibilityAction";
+            }
+
+            @Override
+            public void dispatch(RCTEventEmitter rctEventEmitter) {
+                rctEventEmitter.receiveEvent(reactTag, "topAccessibilityAction", event);
+            }
+        });
+
+        int waitTimeMS = 100;
+        uiController.loopMainThreadForAtLeast(waitTimeMS);
```

Comment:
By "just waiting with no explicit minimal time-span", do we mean using something like uiController.loopMainThreadUntilIdle()`?
I tried that but it doesn't work correctly and the timing remains very flakey as the dispatch doesn't appear to be happening on the main thread.
To avoid that, I'm using a minimum time span here, which feels small enough that it shouldn't be a nuisance in tests.

Reply:
I think this should at least undergo a formalization as a technical debt.

Reply:
> By "just waiting with no explicit minimal time-span", do we mean using something like uiController.loopMainThreadUntilIdle()`? I tried that but it doesn't work correctly and the timing remains very flakey as the dispatch doesn't appear to be happening on the main thread. To avoid that, I'm using a minimum time span here, which feels small enough that it shouldn't be a nuisance in tests.

Yep that's what I meant. Admittedly - it wasn't so clear. I was suggesting that ultimately, via the internal call to `rctEventEmitter.receiveEvent()`, the event either gets handled by the JS or is added to its thread-queue; Either way, those are resources that are monitored by Detox and therefore should be detected as busy in the new loop-for-idle call (hence, no minimal time is required).

Nevertheless I'm fine with keeping it for now - but there's definitely some smell to it.